manifests/diff/diff.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+
+def format_diff(diff: str, path: str) -> str:
+    return f"""<details>
+  <summary><b>{path} DIFF</b></summary>
+
+```diff
+{diff}
+```
+</details>"""
+
+
+def write_to_github_output(
+    diff_output_prod: str, diff_output_dev: str, github_output: str
+):
+    with open(github_output, "a") as f:
+        f.write(f"""prod<<EOF
+{diff_output_prod}
+EOF
+dev<<EOF
+{diff_output_dev}
+EOF
+""")
+
+
+def run_diff(file_path: Path, inputs_ref: str) -> str:
+    result = subprocess.run(
+        [
+            "skipctl",
+            "manifests",
+            "diff",
+            "-p",
+            str(file_path),
+            "--ref",
+            inputs_ref,
+            "--diff-format",
+            "patch",
+        ],
+        capture_output=True,
+        text=True,
+        check=False,
+    )
+    return result.stdout.strip()
+
+
+def generate_diff_output(pattern: str, inputs_path: str, inputs_ref: str) -> str:
+    base_path = Path(inputs_path)
+    matching_dirs = sorted(
+        d for d in base_path.iterdir() if d.is_dir() and d.name.endswith(pattern)
+    )
+
+    output = []
+    for dir_path in matching_dirs:
+        for file_path in sorted(dir_path.rglob("*")):
+            if not file_path.is_file():
+                continue
+
+            try:
+                diff_result = run_diff(file_path, inputs_ref)
+                if diff_result:
+                    display_path = str(file_path.relative_to(inputs_path))
+                    output.append(format_diff(diff_result, display_path))
+            except Exception:
+                pass
```

Comment:
We're silently ignoring errors here too, and might want to propagate the errors so that the action prints as much information as possible, but also fails at the end and blocking PR merge.